dbt/adapters/snowflake/connections.py

```diff
@@ -48,7 +48,11 @@
 
 logger = AdapterLogger("Snowflake")
 _TOKEN_REQUEST_URL = "https://{}.snowflakecomputing.com/oauth/token-request"
-ROW_VALUE_REGEX = re.compile(r"Row Values: \[(.|\n)*\]")
+
+ERROR_REDACTION_PATTERNS = {
+    "Row Values: [redacted]": re.compile(r"Row Values: \[(.|\n)*\]"),
+    "Duplicate field key '[redacted]'": re.compile(r"Duplicate field key '(.|\n)*'")
```

Comment:
To pass the automated code formatting check:

```suggestion
    "Duplicate field key '[redacted]'": re.compile(r"Duplicate field key '(.|\n)*'"),
```